File: app/src/main/java/org/tesira/civic/db/Purchase.java
package org.tesira.civic.db;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "purchases", indices = {@Index("name")})
public class Purchase {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "name")
    private String mName;

    public Purchase(@NonNull String mName) {
        this.mName = mName;
    }

    public String getName() {
        return mName;
    }

}

File: app/src/main/kotlin/org/tesira/civic/db/Purchase.kt
package org.tesira.civic.db // Selbes Paket

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "purchases",
    indices = [Index("name")]
)
data class Purchase(

    @PrimaryKey
    @ColumnInfo(name = "name")
    val name: String
)